scripts/cbmc-viewer/tags.py

```diff
@@ -0,0 +1,166 @@
+"""Support linking a symbol to a line of code defining the symbol.
+
+Run etags over the source tree to produce a TAGS file, parse the TAGS
+file, and construct a mapping from a symbol to a line of code defining
+the symbol.
+
+Warning: In source with conditional compilation or otherwise one
+symbol declared in multiple namespaces (eg type vs function) this
+simple trick may fail.  And it does fail a lot.
+"""
+from __future__ import print_function
+
+from builtins import chr
+from builtins import object
+import os
+import re
+import sys
+import errno
+from past.builtins import cmp
+
+################################################################
+
+
+class Tags(object):
+    """Link a symbol to a line of code defining the symbol."""
+
+    def __init__(self, srcdir=".", tagscmd="etags", tagsfile="TAGS",
+                 fltr=""):
+        """Initialize by running tags over source tree."""
+        self.define = {}
+        self.function = {}
+        self.variable = {}
+        self.make_tags_file(srcdir, tagscmd, tagsfile, fltr)
```

Comment:
This function seems capable to raise exceptions that are not handled anywhere, therefore crashing the program. Is that by design?

Reply:
We have added a generic exception handler to cbmc-viewer. We have added improved exception handling to the TODO.md list of things to fix in the future.